File: apps/Offline-AI-Buddy/android/core/src/main/java/com/americangroupllc/offlineaibuddy/core/models/Language.kt
package com.americangroupllc.offlineaibuddy.core.models

import kotlinx.serialization.Serializable

@Serializable
enum class Language(val displayName: String, val localeIdentifier: String) {
    EN("English", "en-US"),
    HI("हिन्दी", "hi-IN"),
    ZH("中文", "zh-CN"),
    FR("Français", "fr-FR"),
    ES("Español", "es-ES"),
}

File: apps/Offline-AI-Buddy/shared/BuddyAICore/Sources/BuddyAICore/Models/Language.swift
import Foundation

/// The 5 chat languages v1 ships. Adding a 6th is one new case here +
/// one PromptTemplates entry + one ContentPolicy regex set.
public enum Language: String, Codable, CaseIterable, Sendable {
    case en
    case hi
    case zh
    case fr
    case es

    public var displayName: String {
        switch self {
        case .en: return "English"
        case .hi: return "हिन्दी"
        case .zh: return "中文"
        case .fr: return "Français"
        case .es: return "Español"
        }
    }

    /// Maps to BCP-47 locale identifiers used by AVSpeechSynthesizer +
    /// SFSpeechRecognizer + Android Locale.
    public var localeIdentifier: String {
        switch self {
        case .en: return "en-US"
        case .hi: return "hi-IN"
        case .zh: return "zh-CN"
        case .fr: return "fr-FR"
        case .es: return "es-ES"
        }
    }
}
